### ISSUE TITLE
{ error: timeout } bug; likely related to GS' server-side HTTP call timeout handling

### ISSUE THREAD

**Issue by dylanh724** [Imperium42/gs-csharp-community-sdk#7]
**TL;DR**
Occasionally, GS server-side will wait an entire timeout (~30s) before returning an unhandleable (and vague) `{error:timeout}`, either not processing (or partially processing) the script at all. End-user must wait 30s for a response that can't even be handled since there is no additional info provided (except in server-side logs).
![image](https://user-images.githubusercontent.com/8840024/91153700-24f99800-e6f3-11ea-9dff-6caae54c1d04.png)

**Describe the bug**
* Originally thought to be caused by GS WebSocket ( https://github.com/Imperium42/gs-csharp-community-sdk/issues/4 ), this is no longer the case - this is now confirmed to be completely unrelated.

* Users will _rarely_, but _commonly enough to see it more often than you'd want_, see an {error : timeout} in logs that will NOT show in the API call (because it never arrived). 

* It's also confirmed to NOT be an SDK issue, but a GS server-side issue: As I've personally experienced this before when testing via _test harness_. I even have a video somewhere to confirm that I'll post later. 

* This error seemingly _does_ have a pattern: **GS HTTP calls.** Although nothing will appear in the API call history, a log is created and traces back to an HTTP call from annoying sites sometimes don't return either success or an error. This bug bypasses GS' undocumented HTTP timeout parameters. 
  * For example, the HTTP call will still expire the entire script after 30s (even if you set the timeout to 3s):
  ![image](https://user-images.githubusercontent.com/8840024/91152654-d5669c80-e6f1-11ea-8f35-6e6118cc7118.png)

**You can imagine how destructive this bug can be if a crucial HTTP call was "skipped" due to this form of timeout, such as oAuth2 or microtransactions.**

**To Reproduce**
1. Clients call GS (or make a call from test harness) that calls a 3rd-party server (HTTP GET/POST).
2. If the server does not return either a success or error (returns 'nothing', leaving the call unhandled), sometimes GS will ignore any HTTP timeouts (the undocumented timeout screenshot above) and just freeze, timing out completely .
  * Note that sometimes the undocumented timeout (eg, 3 seconds) will work fine, but not always.
3. GS vaguely returns `{ error: timeout }` ~30s later in a way that is _completely unhandleable_.
4. If using test harness, see above screenshot for the res (a near-empty json obj) that is also unhandleable).
5. We cannot do anything because it's deducted to 100% be a GS server.

**Expected behavior**
You should get `{ someRealJsonObj : "Success or error here to handle an expected return" }`

**Screenshots**
1. Rare screenshot of the happening on both client and server-side (from test harness):
![image](https://user-images.githubusercontent.com/8840024/91152036-04304300-e6f1-11ea-9328-78862c870772.png)
^ Within test harness, the script would just freeze for 30s to timeout, then return just `{ "@class" : .ClassName" }` and nothing else.

**Unity Version:**
2020.1, but has been doing this since 5.5~5.6 (forever).

**Additional context**
Since this is server-side, we're sitting ducks on this issue without @Amazon / @GameSparks ' assistance.

**Comment by dylanh724** [comment]
Closing this in favor of https://github.com/Imperium42/gs-csharp-community-sdk/issues/4 and https://github.com/Imperium42/gs-csharp-community-sdk/issues/5
